### PR TITLE
Adding a check-manifest GHA and updating dependent files

### LINKED ISSUE #58: Add GHA MANIFEST checking
See 

- https://github.com/bjlittle/geovista/blob/main/.github/workflows/ci-manifest.yml
- https://packaging.python.org/en/latest/guides/using-manifest-in/

### PR DESCRIPTION
**Pull request by HGWright**
Added a check-manifest GHA and updated the MANIFEST.in file and the pyproject.toml to match it.

closes #58 & #59

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+include *.ipynb
+include *.md
+include *.png
+include *.yaml
+include *.yml
+include .flake8
+include .coveragerc
+include Dockerfile
+recursive-include requirements *.yml
+recursive-include stratify *.pyx
```
Comment (bjlittle) [inline MANIFEST.in:14]: @HGWright This is my suggestion:
```shell
prune .github
exclude .cirrus.yml
exclude .coveragerc
exclude .flake8
exclude .gitignore
exclude .pre-commit-config.yaml
exclude CONTRIBUTING.md
exclude Dockerfile
exclude index.ipynb
include summary.png
recursive-include requirements *.yml *.lock
recursive-include stratify *.pyx
```
What do you think?

**Review by bjlittle — CHANGES_REQUESTED** [review]
@HGWright This is great, thanks for tidying this up :+1: 

However, I've got one suggestion up for discussion...

File: MANIFEST.in (hunk 1 above)
Comment (HGWright) [reply MANIFEST.in]: Looks good to me @bjlittle. I have updated the file with the changes.

File: MANIFEST.in [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+exclude .cirrus.yml
+exclude .coveragerc
+exclude .flake8
+exclude .gitignore
+exclude .pre-commit-config.yaml
+exclude CONTRIBUTING.md
+exclude Dockerfile
+exclude index.ipynb
+include summary.png
+recursive-include requirements *.yml *.lock
```
Comment (bjlittle) [inline MANIFEST.in:11]:
```suggestion
recursive-include requirements *.yml *.lock *.md
```

**Review by bjlittle — APPROVED** [review]
@HGWright Nice one, this looks good to me :smile:

**Pull request merged**